app/src/main/java/com/mailerdaemon/app/placement/PlacementActivity.kt

```diff
@@ -11,51 +11,50 @@ import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.SearchView
 import androidx.recyclerview.widget.DiffUtil
 import com.mailerdaemon.app.ApplicationClass
+import com.mailerdaemon.app.PlacementFragment
 import com.mailerdaemon.app.R
+import com.mailerdaemon.app.ShowData
 import com.mailerdaemon.app.toast
 import com.mailerdaemon.app.utils.AccessDatabase
 import kotlinx.android.synthetic.main.activity_placement.*
+import kotlinx.android.synthetic.main.fragment_placement.*
+import kotlinx.android.synthetic.main.item_posts.*
 import kotlinx.android.synthetic.main.shimmer_layout_posts.*
 import retrofit2.Call
 import retrofit2.Callback
 import retrofit2.Response
 
 class PlacementActivity : AppCompatActivity(), AccessDatabase {
 
-    private var access: Boolean? = null
-    var data: List<PlacementModel>? = null
+    var data = emptyList<PlacementModel>()
+    val fragment = PlacementFragment()
+    private lateinit var adapter: PlacementAdapter
```

Comment:
Why is Adapter in Activity? It is not showing recyclerView. You are getting confused here. 
You need to pass the data to adapter in Fragment for that you created showData() interface.

Reply:
Below in code you have applied DiffUtil to this Adater of Activity. Please understand what diffUtil does. You need to move it fragment.

Reply:
Diffutil calculates difference in old list and new list and dispatches result to adapter. So it should be called in showData method in frament.